enhancing/enhancer.js

```diff
@@ -1,22 +1,79 @@
+
+//EXPORTS
 module.exports = {
-  succeed,
-  fail,
-  repair,
-  get,
-};
-
-function succeed(item) {
-  return { ...item };
-}
-
-function fail(item) {
-  return { ...item };
-}
-
-function repair(item) {
-  return { ...item };
-}
-
-function get(item) {
-  return { ...item };
-}
+  success: ( item ) => {
+    const itemName = item.unenhancedName;
+
+    if ( item.enhancement <= 14 && item.durability < 25 ) {
+      throw new Error( 'Durability is below 25; item cannot be enhanced' );
+    }
+
+    if ( item.enhancement >= 15 && item.durability < 10 ) {
+      throw new Error( 'Durability is below 10; item can not be enhanced' );
+    }
+
+    if ( item.enhancement === 20 ) {
+      throw new Error( 'Item enhancement is maxed out' );
+    }
+
+    if ( item.enhancement >= 0 && item.enhancement < 15 ) {
+      item.enhancement += 1;
```

Comment:
you could also utilize a switch or chained ternary here as well, not saying any way is correct, just other options for long if/else statements